zipline/protocol.py

```diff
@@ -360,17 +367,27 @@ def minute_get_bars(days, env=None):
         return self._get_bars(days)
 
     def mavg(self, days):
-        return self._get_buffer(self._get_bars(days)).mean()
+        bars = self._get_bars(days)
+        prices = self._get_buffer(bars, raw=True)
+        return nanmean(prices)
 
     def stddev(self, days):
-        return self._get_buffer(self._get_bars(days)).std(ddof=1)
+        bars = self._get_bars(days)
+        prices = self._get_buffer(bars, raw=True)
+        return nanstd(prices, ddof=1)
 
     def vwap(self, days):
         bars = self._get_bars(days)
-        prices = self._get_buffer(bars)
-        vols = self._get_buffer(bars, field='volume')
+        prices = self._get_buffer(bars, raw=True)
+        vols = self._get_buffer(bars, field='volume', raw=True)
+
+        vol_sum = nansum(vols)
+        if vol_sum == 0:
```

Comment:
https://docs.python.org/3.5/glossary.html#term-eafp

Easier to ask forgiveness than permission.